version:  '3'

networks:
  default:
    external:
      name: ${EXTERNAL_NET}

services:
  webapp:
    restart: on-failure
    environment:
      - XDEBUG_IDEKEY=${XDEBUG_IDEKEY}
      - XDEBUG_REMOTE_HOST=${XDEBUG_HOST}
      - XDEBUG_REMOTE_PORT=${XDEBUG_PORT}
      - XDEBUG_REMOTE_CONNECT_BACK=${XDEBUG_CONNECT_BACK}
    build:
      context: .
      dockerfile: docker/Dockerfile
      args:
        TIMEZONE: ${TIMEZONE}
        USER: ${USER}
        USER_NAME: ${USER_NAME}
        USER_EMAIL: ${USER_EMAIL}
        USER_ID: ${USER_ID}
    volumes:
      - ./app/:/var/www/app

  nginx:
    container_name: ${APP}
    image: nginx:1.15.3-alpine
    restart: on-failure
    volumes:
      - ./app/:/var/www/app
      - ./docker/nginx/default.conf:/etc/nginx/conf.d/default.conf:ro
      - ./docker/nginx/logs:/var/log/nginx
    ports:
      - ${NGINX_PORT}:80
    depends_on:
      - webapp
